### PR TITLE
Add document for column-based storage

### PR DESCRIPTION
**Pull request by ButterBright**

### CODE REVIEW

File: docs/concept/tsdb.md [hunk 1]
```diff
@@ -2,18 +2,37 @@
...
+
+[segment](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/segment.png)
+
+## Part
+
+Within a part, data is split into multiple files in a columnar manner. The timestamps are stored in the `timestamps.bin` file, tags are organized as multiple files with the `.tf` suffix using column families, and fields are stored separately in the `fields.bin` file. 
+
+In addition, each part maintains several metadata files. Among them, `metadata.json` is the metadata file for the part, storing descriptive information about the part. The `meta.bin` file serves as the entry file for the entire part, helping to index the `primary.bin` file. Through the `primary.bin` file, the actual data files or the tagFamily metadata files ending with `.tfm` can be indexed, which in turn helps locate the tag data.
```
Comment (hanahmily) [inline docs/concept/tsdb.md:23]:
```suggestion
In addition, each part maintains several metadata files. Among them, `metadata.json` is the metadata file for the part, storing descriptive information, such as start and end times, part size and etc. 

The `meta.bin` is a skipping index file serves as the entry file for the entire part, helping to index the `primary.bin` file. 
The `primary.bin` file contains the index of each block. Through it, the actual data files or the tagFamily metadata files ending with `.tfm` can be indexed, which in turn helps locate the data in blocks. 
```

File: docs/concept/tsdb.md [hunk 2]
```diff
@@ -2,18 +2,37 @@
...
+## Part
+
+Within a part, data is split into multiple files in a columnar manner. The timestamps are stored in the `timestamps.bin` file, tags are organized as multiple files with the `.tf` suffix using column families, and fields are stored separately in the `fields.bin` file. 
+
+In addition, each part maintains several metadata files. Among them, `metadata.json` is the metadata file for the part, storing descriptive information about the part. The `meta.bin` file serves as the entry file for the entire part, helping to index the `primary.bin` file. Through the `primary.bin` file, the actual data files or the tagFamily metadata files ending with `.tfm` can be indexed, which in turn helps locate the tag data. 
+
+Notably, for data of the `Stream` type, since there are no field columns, the `fields.bin` file does not exist, while the rest of the structure is entirely consistent with the `Measure` type.
+
+[measure-part](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/measure-part.png)
+[stream-part](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/stream-part.png)
```
Comment (hanahmily) [inline docs/concept/tsdb.md:28]:
```suggestion
![measure-part](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/measure-part.png)
![stream-part](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/stream-part.png)
```

File: docs/concept/tsdb.md [hunk 3]
```diff
@@ -2,18 +2,37 @@
...
+
+## Block
 
-[shard](https://skywalking.apache.org/doc-graph/banyandb/v0.4.0/tsdb-shard.png)
+The figure below shows the detailed fields within each block. It is worth noting that when storing tag and field data, BanyanDB prioritizes storing data points or elements with the same tag consecutively. This helps improve the compression efficiency of the files.
```
Comment (hanahmily) [inline docs/concept/tsdb.md:32]:
```suggestion
The diagram below shows the detailed fields within each block. The block is the minimal unit of tsdb, which contains several rows of data. Due to the column-base design, each block is spread over several files.
```

File: docs/concept/tsdb.md [hunk 4]
```diff
@@ -2,18 +2,37 @@
...
 
-* Buffer: It is typically implemented as an in-memory queue managed by a shard. When new time-series data is ingested into the system, it is added to the end of the queue, and when the buffer reaches a specific size, the data is flushed to disk in batches.
-* SST: When a bucket of buffer becomes full or reaches a certain size threshold, it is flushed to disk as a new Sorted String Table (SST) file. This process is known as compaction.
-* Segments and Blocks: Time-series data is stored in data segments/blocks within each shard. Blocks contain a fixed number of data points and are organized into time windows. Each data segment includes an index that efficiently retrieves data within the block.
-* Block Cache: It manages the in-memory cache of data blocks, improving query performance by caching frequently accessed data blocks in memory.
+[measure-block](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/measure-block.png)
+[stream-block](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/stream-block.png)
```
Comment (hanahmily) [inline docs/concept/tsdb.md:35]:
```suggestion
![measure-block](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/measure-block.png)
![stream-block](https://skywalking.apache.org/doc-graph/banyandb/v0.6.0/stream-block.png)
```

**Review by hanahmily — CHANGES_REQUESTED** [review]

File: docs/concept/tsdb.md [hunk 5]
```diff
@@ -2,37 +2,58 @@
...
 
-Once a bucket is closed, it is stored as a single SST in a shard. The file is indexed and added to the index for the corresponding time range and resolution.
+Whenever a new memory part is generated or a flush and merge operation is triggered, it initiates an update of the snapshot and deletes outdated snapshots.
 
 ## Read Path
 
-The read path in TSDB retrieves time-series data from disk or memory and returns it to the query engine. The read path comprises several components: the buffer, cache, and SST file. The following is a high-level overview of how these components work together to retrieve time-series data in TSDB.
-
-The first step in the read path is to perform an index lookup to determine which blocks contain the desired time range. The index contains metadata about each data block, including its start and end time and its location on disk.
+The read path in TSDB retrieves time-series data from disk or memory and returns it to the query engine. The read path comprises several components: the buffer and parts. The following is a high-level overview of how these components work together to retrieve time-series data in TSDB.
```
Comment (Superskyyy) [inline docs/concept/tsdb.md:53]:
```suggestion
The read path in TSDB retrieves time-series data from disk or memory, and returns it to the query engine. The read path comprises several components: the buffer and parts. The following is a high-level overview of how these components work together to retrieve time-series data in TSDB.
```

**Review by Superskyyy — CHANGES_REQUESTED** [review]
grammar nits.

File: docs/concept/tsdb.md [hunk 6]
```diff
@@ -2,37 +2,58 @@
...
 
-When a shard receives a write request, the data is written to the buffer as a series of buckets. Each bucket is a fixed-size chunk of time-series data typically configured to be several minutes or hours long. As new data is written to the buffer, it is appended to the current bucket until it is full. Once the bucket is full, it is closed, and a new bucket is created to continue buffering writes.
+When a shard receives a write request, the data is written to the buffer as a memory part. Meanwhile, the series index and inverted index will also be updated. The worker in the background periodically flushes data, writing the memory part to the disk. After the flush operation is completed, it triggers a merge operation to combine the parts and remove invalid data. 
 
-Once a bucket is closed, it is stored as a single SST in a shard. The file is indexed and added to the index for the corresponding time range and resolution.
+Whenever a new memory part is generated, or when a flush or merge operation is triggered, they initiate an update of the snapshot and delete outdated snapshots.
```
Comment (hanahmily) [inline docs/concept/tsdb.md:49]:
```suggestion
Whenever a new memory part is generated, or when a flush or merge operation is triggered, they initiate an update of the snapshot and delete outdated snapshots. The parts in a persistent snapshot could be accessible to the reader.
```

**Review by hanahmily — CHANGES_REQUESTED** [review]

**Comment by hanahmily** [comment]
All diagrams look good to me.

**Comment by wu-sheng** [comment]
> All diagrams look good to me.

Should we merge the website and make sure the static path generated?

**Comment by hanahmily** [comment]
> > All diagrams look good to me.
> 
> Should we merge the website and make sure the static path generated?

Sure, go for it.

**Review by hanahmily — DISMISSED** [review]

**Comment by ButterBright** [comment]
> > ERROR: 1 dead links found!
> > [✖] https://ubuntu.com/wsl → Status: 0
> 
> But this website seems to redirect to `https://ubuntu.com/desktop/wsl`. Docs may need a update?

Let me fix these links.

**Review by wu-sheng — APPROVED** [review]

**Comment by Superskyyy** [comment]
The images are kind of coupled with the website, if someone can't really access website url in their offline local doc the images are not loadable right?

**Comment by wu-sheng** [comment]
> The images are kind of coupled with the website, if someone can't really access website url in their offline local doc the images are not loadable right?

No, they are not. But the good part is, the size of the git repo of source codes are very limited, this would be helpful for the developers.

**Review by Superskyyy — APPROVED** [review]
LGTM

**Pull request merged**